### PR TITLE
Reproduce goal representations and steering in Procgen Maze

### LINKED ISSUE #58: [Notebook] Reproduce goal representations and steering in Procgen Maze
Parent roadmap: #60  
Library dependencies: #52, #53, and #54

## Paper

[Understanding and Controlling a Maze-Solving Policy Network](https://arxiv.org/abs/2310.08043)  
Reference code: [UlisseMini/procgen-tools](https://github.com/UlisseMini/procgen-tools)

## Outcome

Deliver `docs/source/reproductions/maze-policy-goal-representations.ipynb`, a dedicated scientific reproduction notebook separate from tutorials.

For one immutable pretrained policy and frozen maze set, the notebook should:

1. recover the goal-location representation analysis for the reported channels;
2. test redundancy/distribution using declared channel controls; and
3. reproduce one steering intervention, measuring policy-distribution and behavioural effects.

## Notebook requirements

- Pin the paper version, procgen-tools revision, checkpoint digest/license status, Procgen version, maze seeds, and adapter mapping.
- Demonstrate parameter and uninstrumented-output parity with the reference policy.
- Predeclare channel/layer selection and evaluate on a held-out split with random/shuffled-channel controls.
- Use #54 for matched baseline/control/intervention workflows and #53 for external/result artifact identities.
- Report vector-field or equivalent quantitative policy changes, behavioural effects, failures, and uncertainty.
- Distinguish correlation, causal contribution, controllability, and statements about policy goals.
- Include a bounded smoke path clearly marked as non-scientific.

## Acceptance criteria

- [ ] The dedicated notebook renders from the Reproductions gallery introduced by #52.
- [ ] The XDRL execution path uses public XDRL/TDHook interfaces rather than procgen-tools hook state.
- [ ] Baseline and intervention runs share the frozen checkpoint and maze set.
- [ ] Reference comparison and all material implementation deviations are explicit.
- [ ] Claims remain limited to the checkpoint and maze distribution evaluated.

## Non-goals

This issue does not port the full procgen-tools codebase or infer general goal representations.

### PR DESCRIPTION
**Pull request by Xmaster6y**
## Summary

- add a dedicated Procgen Maze goal-representation reproduction notebook to the Reproductions gallery
- predeclare the reported/effective channels, held-out split, shuffled-channel and shuffled-label controls, uncertainty, and matched steering metrics
- use public XDRL/TDHook interaction, intervention, and provenance interfaces with explicit adapter parity checks
- fail closed in paper mode because the advertised checkpoint URL is unavailable and the checkpoint lacks an immutable digest/license and frozen reference result bundle
- keep the default bounded synthetic smoke path explicitly non-scientific

## Validation

- notebook smoke execution via `jupyter nbconvert --execute`
- paper mode fails closed with the declared asset error
- `uv run pre-commit run --all-files`
- `just tests` (236 passed, 95.54% coverage)
- `just docs` (warning-strict Sphinx build)
- `git diff --cached --check`

Closes #58

### CODE REVIEW

File: docs/source/reproductions/maze-policy-goal-representations.ipynb [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+    "def desired_action(positions, target):\n",
+    "    row_delta = target[0] - positions[:, 0]\n",
+    "    column_delta = target[1] - positions[:, 1]\n",
+    "    return torch.where(\n",
+    "        row_delta < 0,\n",
+    "        torch.zeros_like(row_delta),\n",
+    "        torch.where(\n",
+    "            row_delta > 0,\n",
+    "            torch.ones_like(row_delta),\n",
+    "            torch.where(column_delta < 0, torch.full_like(row_delta, 2), torch.full_like(row_delta, 3)),\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/maze-policy-goal-representations.ipynb]: Fixed in 0cbdaaa. `desired_action` now returns a validity mask, and the target-position row is excluded from `target_directed_probability_delta`. The metric is also restricted to held-out levels.

[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/maze-policy-goal-representations.ipynb]: Fixed in 0cbdaaa. Zero-distance positions now receive `valid=False` and are masked from the target-directed probability calculation; only positions with an actual target-directed move contribute.

File: docs/source/reproductions/maze-policy-goal-representations.ipynb [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+    "\n",
+    "\n",
+    "def run_arm(label, channels, coordinate):\n",
+    "    callback = patch_at(channels, coordinate)\n",
+    "    pair = runner.run_paired(\n",
+    "        Workflow(SteeringVectors([WORKFLOW_TARGET], steer_fn=no_op)),\n",
+    "        Workflow(SteeringVectors([WORKFLOW_TARGET], steer_fn=callback)),\n",
+    "        batch,\n",
+    "        pair_id=f\"maze-goal-representations:smoke:{label}\",\n",
+    "        code_revision=\"working-tree-smoke\",\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/maze-policy-goal-representations.ipynb]: Fixed in 0cbdaaa. The notebook now resolves `git rev-parse HEAD`, appends `+dirty` when tracked files differ, passes that value as `code_revision`, and exposes it in the verdict. A clean post-commit execution recorded `0cbdaaaf178abc6bcbde55c8ff5f6d662a2cea38`.

File: docs/source/reproductions/maze-policy-goal-representations.ipynb [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+    "    generator = torch.Generator().manual_seed(SEED + 303)\n",
+    "    values = []\n",
+    "    for _ in range(draws):\n",
+    "        indices = torch.randint(len(evaluation_levels), (len(evaluation_levels),), generator=generator)\n",
+    "        values.append(float((changed[indices] - baseline[indices]).mean()))\n",
+    "    return [float(value) for value in torch.tensor(values).quantile(torch.tensor((0.025, 0.975)))]\n",
+    "\n",
+    "\n",
+    "target_actions = desired_action(bundle[\"positions\"], INTERVENTION_TARGET).repeat(32)\n",
+    "fixed_steering_targets = torch.tensor(INTERVENTION_TARGET).repeat(32, 1)\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/maze-policy-goal-representations.ipynb]: Fixed in 0cbdaaa. Each metric arm now reads its own target coordinate from `intervention_arms`; directed-action labels and greedy target rollouts are derived from that coordinate. The reported result includes `arm_target` for auditability.

File: docs/source/reproductions/maze-policy-goal-representations.ipynb [hunk 4]
```diff
@@ -0,0 +1,679 @@
...
+    "        named_tensor_digest(FROZEN_LEVEL_TENSORS),\n",
+    "        metadata={\n",
+    "            \"level_ids\": bundle[\"level_ids\"].tolist(),\n",
+    "            \"evaluation_level_ids\": bundle[\"level_ids\"][evaluation_levels].tolist(),\n",
+    "            \"frozen\": True,\n",
+    "        },\n",
+    "    ),\n",
+    "    InputArtifactReference(\n",
+    "        \"reference:arxiv-2310.08043v1\",\n",
+    "        InputArtifactRole.REFERENCE_RESULT,\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/maze-policy-goal-representations.ipynb]: Fixed in 0cbdaaa. The paper PDF is now `InputArtifactRole.OTHER`, identified as `paper:arxiv-2310.08043v1`, with metadata explicitly stating `artifact_kind=paper_pdf` and `reference_results_available=false`.
